ledfx/api/virtual_tools.py

```diff
@@ -49,22 +96,18 @@ async def put(self, request: web.Request) -> web.Response:
                     virtual.force_frame(parse_color(validate_color(color)))
 
         if tool == "oneshot":
-            color = parse_color(validate_color(data.get("color", "white")))
-            ramp = data.get("ramp", 0)
-            hold = data.get("hold", 0)
-            fade = data.get("fade", 0)
-            brightness = min(1, max(0, data.get("brightness", 1)))
-
-            # if all values are zero, we will now just ensure any current
-            # oneshot are cancelled
-
-            # iterate through all virtuals and apply oneshot
+            # Disable all oneshot Flash if put request is sent.
+            result = False
             for virtual_id in self._ledfx.virtuals:
                 virtual = self._ledfx.virtuals.get(virtual_id)
                 if virtual is not None:
-                    virtual.add_oneshot(
-                        Flash(color, ramp, hold, fade, brightness)
-                    )
+                    for oneshot in virtual.oneshots:
+                        if isinstance(oneshot, Flash):
+                            oneshot.active = False
```

Comment:
As the comment for virtual_tools.py instance, I think this will lead to the instance being orphaned in the list with cumulative cruft, rather than discarding from the list for garbage collection.